implementations/micrometer-registry-statsd/src/test/java/io/micrometer/statsd/StatsdMeterRegistryTest.java

```diff
@@ -312,17 +312,52 @@ void slasOnlyNoPercentileHistogram(StatsdFlavor flavor) {
 
         Gauge summaryHist1 = registry.get("my.summary.histogram").tags("le", "1").gauge();
         Gauge summaryHist2 = registry.get("my.summary.histogram").tags("le", "2").gauge();
-        Gauge timerHist = registry.get("my.timer.histogram").tags("le", "1").gauge();
+        Gauge summaryHist3 = registry.get("my.summary.histogram").tags("le", "+Inf").gauge();
+        Gauge timerHist1 = registry.get("my.timer.histogram").tags("le", "1").gauge();
+        Gauge timerHist2 = registry.get("my.timer.histogram").tags("le", "+Inf").gauge();
 
         assertThat(summaryHist1.value()).isEqualTo(1);
         assertThat(summaryHist2.value()).isEqualTo(1);
-        assertThat(timerHist.value()).isEqualTo(1);
+        assertThat(summaryHist3.value()).isEqualTo(1);
+        assertThat(timerHist1.value()).isEqualTo(1);
+        assertThat(timerHist2.value()).isEqualTo(1);
 
         clock.add(config.step());
 
         assertThat(summaryHist1.value()).isEqualTo(0);
         assertThat(summaryHist2.value()).isEqualTo(0);
-        assertThat(timerHist.value()).isEqualTo(0);
+        assertThat(summaryHist3.value()).isEqualTo(0);
+        assertThat(timerHist1.value()).isEqualTo(0);
+        assertThat(timerHist2.value()).isEqualTo(0);
+    }
+
+    @Test
+    void timersWithSlasHaveInfBucket() {
+        StatsdMeterRegistry registry = new StatsdMeterRegistry(configWithFlavor(StatsdFlavor.ETSY), clock);
+        Timer timer = Timer.builder("my.timer").sla(Duration.ofMillis(1)).register(registry);
+
+        assertThat(registry.get("my.timer.histogram").tag("le", "+Inf").gauge()).isNotNull();
+    }
+
+    @Test
+    void distributionSummariesWithSlasHaveInfBucket() {
+        StatsdMeterRegistry registry = new StatsdMeterRegistry(configWithFlavor(StatsdFlavor.ETSY), clock);
+        DistributionSummary summary = DistributionSummary.builder("my.distribution").sla(1).register(registry);
+
+        assertThat(registry.get("my.distribution.histogram").tag("le", "+Inf").gauge()).isNotNull();
```

Comment:
Can remove the `assertThat` because `get` will throw a detailed exception if a matching gauge isn't found.

Reply:
Done. Thanks!